FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/HandlePasswordProtectedMsgWorker.kt

```diff
@@ -0,0 +1,425 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.jetpack.workmanager
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.flowcrypt.email.R
+import com.flowcrypt.email.api.email.EmailUtil
+import com.flowcrypt.email.api.email.JavaEmailConstants
+import com.flowcrypt.email.api.retrofit.FlowcryptApiRepository
+import com.flowcrypt.email.api.retrofit.request.model.MessageUploadRequest
+import com.flowcrypt.email.database.MessageState
+import com.flowcrypt.email.database.entity.AccountEntity
+import com.flowcrypt.email.database.entity.MessageEntity
+import com.flowcrypt.email.extensions.javax.mail.internet.getAddresses
+import com.flowcrypt.email.extensions.javax.mail.internet.getFromAddress
+import com.flowcrypt.email.extensions.javax.mail.internet.getMatchingRecipients
+import com.flowcrypt.email.extensions.kotlin.toInputStream
+import com.flowcrypt.email.jetpack.viewmodel.AccountViewModel
+import com.flowcrypt.email.jetpack.workmanager.base.BaseMsgWorker
+import com.flowcrypt.email.security.KeyStoreCryptoManager
+import com.flowcrypt.email.security.KeysStorageImpl
+import com.flowcrypt.email.security.pgp.PgpDecryptAndOrVerify
+import com.flowcrypt.email.security.pgp.PgpEncryptAndOrSign
+import com.flowcrypt.email.util.GeneralUtil
+import com.flowcrypt.email.util.LogsUtil
+import com.flowcrypt.email.util.exception.ExceptionUtil
+import com.flowcrypt.email.util.google.GoogleApiClientHelper
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.gson.GsonBuilder
+import com.sun.mail.util.MailConnectException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+import org.bouncycastle.openpgp.PGPPublicKeyRingCollection
+import org.bouncycastle.openpgp.PGPSecretKeyRingCollection
+import org.pgpainless.util.Passphrase
+import java.io.ByteArrayOutputStream
+import java.io.FileNotFoundException
+import java.io.InputStream
+import java.net.SocketException
+import java.util.Base64
+import java.util.Properties
+import javax.mail.Message
+import javax.mail.MessagingException
+import javax.mail.Multipart
+import javax.mail.Session
+import javax.mail.internet.InternetAddress
+import javax.mail.internet.MimeBodyPart
+import javax.mail.internet.MimeMessage
+import javax.net.ssl.SSLException
+import kotlin.random.Random
+
+/**
+ * @author Denis Bondarenko
+ *         Date: 12/29/21
+ *         Time: 9:30 AM
+ *         E-mail: DenBond7@gmail.com
+ */
+class HandlePasswordProtectedMsgWorker(context: Context, params: WorkerParameters) :
```

Comment:
@tomholub This class contains logic from https://github.com/FlowCrypt/flowcrypt-ios/issues/1254